File: lc-dc-sep/lc-dc-sep2/solution.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        long long s = 0;
        for(int itr: chalk) {
            s += itr;
        }
        int t = k % s;
        if(t > 0){
            for(int i=0; i<chalk.size(); i++){
                int n = chalk.at(i);
                if(n <= t) t -= n;
                else return i;
            }
        }
        return 0;
    }
};

File: lc-dc-sep/lc-dc-sep2/solution.py
# Problem: 1894. Find the Student that Will Replace the Chalk

class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        s = sum(chalk)
        t = k % s

        if t > 0:
            for i, n in enumerate(chalk):
                if n <= t:
                    t -= n
                else:
                    return i

        return 0
        